test/controller/objecting.entity.name.controller.spec.unit.ts

```diff
@@ -33,25 +36,107 @@ const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
 mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next());
 
 const mockSessionMiddleware = sessionMiddleware as jest.Mock;
-mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
-  req.session = {
-    data: {},
-  } as Session;
-  return next();
-});
 
 const mockObjectionSessionMiddleware = objectionSessionMiddleware as jest.Mock;
-mockObjectionSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
-  if (req.session) {
-    req.session.data[OBJECTIONS_SESSION_NAME] = {};
-    return next();
-  }
 
-  return next(new Error("No session on request"));
-});
+const mockRetrieveFromObjectionSession = retrieveFromObjectionSession as jest.Mock;
+
+const mockRetrieveObjectionSessionFromSession = retrieveObjectionSessionFromSession as jest.Mock;
+
+const mockGetObjection = getObjection as jest.Mock;
 
 describe("objecting entity name tests", () => {
 
+  beforeEach(() => {
+    mockSessionMiddleware.mockReset();
+    mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+      req.session = {
+        data: {},
+      } as Session;
+      return next();
+    });
+    mockObjectionSessionMiddleware.mockReset();
+    mockObjectionSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+      if (req.session) {
+        req.session.data[OBJECTIONS_SESSION_NAME] = {};
+        return next();
+      }
+
+      return next(new Error("No session on request"));
+    });
+  });
+
+  it("should render empty objecting entity name page if no change answer flag is set in session", async() => {
+    mockRetrieveFromObjectionSession.mockReset();
+    mockRetrieveFromObjectionSession.mockReturnValueOnce(undefined);
+    const response = await request(app).get(OBJECTIONS_OBJECTING_ENTITY_NAME)
```

Comment:
could do with some whitespace to separate mocks from the 'get' call and from the expects.

Reply:
Added spaces.